lmm:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  joint_1_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_1_1
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_1_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_1_2
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_1_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_1_3
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_2_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_2_1
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_2_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_2_2
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_2_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_2_3
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_3_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_3_1
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_3_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_3_2
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_3_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_3_3
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_4_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_4_1
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_4_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_4_2
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_4_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_4_3
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_5_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_5_1
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_5_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_5_2
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_5_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_5_3
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_6_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_6_1
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_6_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_6_2
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_6_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_6_3
    pid: {p: 10, i: 0.001, d: 0.1}

  joint_m_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_m_1
    pid: {p: 10, i: 0.01, d: 1}

  joint_m_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_m_2
    pid: {p: 10, i: 0.01, d: 1}

  joint_m_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_m_3
    pid: {p: 10, i: 0.01, d: 1}

  joint_m_4_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint_m_4
    pid: {p: 1, i: 0.0001, d: 0.01}
